Add a hybrid example related to threads

0152la opened the pull request:
Check whether we can set DDCs for spawned threads individually, and independent from the main program thread.

ltratt commented on hybrid/threads_ddc.c:
```diff
@@ -0,0 +1,69 @@
+    for (size_t i = 0; i < pow(10, 6); ++i)
+    {
+    };
+    curr_t_info->ddc_out = read_ddc();
```
Can we add an `assert(curr_t_info->ddc_out == curr_t_info->addr_in)` or similar here?

0152la replied:
Done in https://github.com/capablevms/cheri-examples/pull/94/commits/66126d788c2d2ad8c04c8f499cd590f7b9a93359. Also added a check to ensure the DDC is changed.

ltratt commented:
Please squash.

0152la commented:
Squashed.

0152la commented:
I've made a bit of an ugly fix in https://github.com/capablevms/cheri-examples/pull/94/commits/42f7b6ce994e5f76cc682485f6c16f545715e92c, but I think a proper fix would involve a bit too much infrastructure changing (at least I can't think of any other more straightforward one to only add the libraries we need to the new test).

0152la commented:
Squashed.

0152la commented:
Is it fine to force-push to get the changes from #95 ?

ltratt commented:
Please force push a rebase.

0152la commented:
Rebased in https://github.com/capablevms/cheri-examples/pull/94/commits/e962618ab5245c8d8706b1339795157098031fe0 (also fixed `-lpthreads` to `-lpthread` in the makefile).
